examples/plot_bch_so3.py

```diff
@@ -19,7 +19,7 @@
 
 import geomstats.backend as gs
 from geomstats.geometry.skew_symmetric_matrices import SkewSymmetricMatrices
-from geomstats.geometry.special_orthogonal import SpecialOrthogonal
+from geomstats.geometry.special_orthogonal3 import SpecialOrthogonal
```

Comment:
Do we want to name it `SpecialOrthogonal3` so that there is no conflict when a user wants to import both `SpecialOrthogonal3` and `SpecialOrthogonal` in the same file?

Or should they always import `SpecialOrthogonal`, and entering `n=3` calls `SpecialOrthogonal3` behind the scenes? (probably preferred?)

Reply:
how would behind the scene work? return self in the `__init__`?

Reply:
Have a `if n == 3` in the SpecialOrthogonal's `__init__` that calls `SpecialOrthogonal3.__init__()`, and `else` that runs the usual constructor's code?

Reply:
calling or returning `SpecialOrthogonal3.__init__()` or `SpecialOrthogonal3.__init__(self)` does not work. I agree with you, I just don't know how to do, maybe a wrapper deciding which class to instantiate?

Reply:
It needs to inherit from SpecialOrthogonal3 in order to work. Then, this syntax should work:
https://github.com/geomstats/geomstats/blob/e44f44a968fbfa963313cc4fe71d326b6f66e9cd/geomstats/geometry/special_orthogonal.py#L60

What fails when you try it?

Reply:
But does it make sense to have `_SpecialOrthogonal3` as a parent class of `SpecialOrthogonal`? And will it actually work with MRO (because SpecialOrthogonal must inherit from LieGroup and GeneralLinear)? (I will try)